src/v/security/license.cc

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "security/license.h"
+
+#include "json/document.h"
+#include "utils/base64.h"
+
+#include <boost/date_time/gregorian/formatters.hpp>
+#include <boost/date_time/gregorian/parsers.hpp>
+#include <boost/filesystem.hpp>
+#include <cryptopp/base64.h>
+#include <cryptopp/rsa.h>
+#include <cryptopp/sha.h>
+
+namespace security {
+
+namespace crypto {
+
+static ss::sstring parse_pem_contents(const ss::sstring& pem_key) {
+    static ss::sstring public_key_header = "-----BEGIN PUBLIC KEY-----";
+    static ss::sstring public_key_footer = "-----END PUBLIC KEY-----";
```

Comment:
static in this context for what looks like constants adds generated extra code and guard instructions to make multi-threaded access safe. that's generally not a property we need in seastar.

Reply:
nice catch, should be `static const`